application/controllers/LimereplacementfieldsController.php

```diff
@@ -37,7 +37,7 @@ public function actionIndex()
             throw new CHttpException(401);
         }
 
-        if (!Permission::model()->hasSurveyPermission($surveyid, 'survey', 'read')) {
+        if ($surveyid && !Permission::model()->hasSurveyPermission($surveyid, 'survey', 'read')) {
```

Comment:
Shouldn't we check this and throw exception prevously, maybe before querying the survye model?

Reply:
No it was added to fix a security issue

It's your commit
https://github.com/LimeSurvey/LimeSurvey/commit/64e2003319ffffdcd91d98cb22d9c48f5e86782b

Reply:
PS : must not be surveycontent/read here ?

Reply:
I was referring to checking the surveyid is not empty. Sorry.

Reply:
> PS : must not be surveycontent/read here ?

Yes, think it is a good idea

Reply:
> I was referring to checking the surveyid is not empty. Sorry.

If surveyid != "" and \Survey not found : must throw a 404 here

But : must throw a 403 only is surveyid != ""  .

If sureyid is not set : return an empty array.

Reply:
> Yes, think it is a good idea

Not related to this fix ;)